src/main/java/seedu/fast/logic/parser/FindCommandParser.java

```diff
@@ -24,10 +26,33 @@ public FindCommand parse(String args) throws ParseException {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
+        if (trimmedArgs.startsWith(PriorityTag.PRIORITY_TAG_PREFIX)) {
+            trimmedArgs = trimmedArgs.substring(
+                    PriorityTag.PRIORITY_TAG_PREFIX.length());
+            String[] tags = trimmedArgs.split("\\s+");
+            for (String tag:tags) {
+                if (isNotPriority(tag)) {
+                    throw new ParseException(
+                            String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+                }
+            }
+            return new FindCommand(new PriorityPredicate(Arrays.asList(tags)));
+        } else {
+            String[] nameKeywords = trimmedArgs.split("\\s+");
+
+            return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        }
+    }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
-
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+    private static boolean isNotPriority(String tag) {
+        switch(tag) {
```

Comment:
I think that a if...else... statement would be more fitting here? Or simple just a 
`return PriorityTag.LowPriority.TERM || PriorityTag.MediumPriority.TERM || PriorityTag.HighPriority.TERM;`

This current implementation defeats the purpose of a switch statement as there are only 2 outcomes.